app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers/MailchimpTags.php

```diff
@@ -0,0 +1,813 @@
+<?php
+
+/**
+ * mailchimp-lib Magento Component
+ *
+ * @category  Ebizmarts
+ * @package   mailchimp-lib
+ * @author    Ebizmarts Team <info@ebizmarts.com>
+ * @copyright Ebizmarts (http://ebizmarts.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+class Ebizmarts_MailChimp_Model_Api_Subscribers_MailchimpTags
+{
+
+    private $_storeId;
+    private $_mailChimpTags;
+    private $_subscriber;
+    private $_customer;
+    private $_mcHelper;
+
+    public function __construct()
+    {
+        $this->setMailChimpHelper();
+    }
+
+    public function setStoreId($storeId)
+    {
+        $this->_storeId = $storeId;
+    }
+
+    public function getStoreId()
+    {
+        return $this->_storeId;
+    }
+
+    public function setSubscriber($subscriber)
+    {
+        $this->_subscriber = $subscriber;
+    }
+
+    public function getSubscriber()
+    {
+        return $this->_subscriber;
+    }
+
+    public function setCustomer($customer)
+    {
+        $this->_customer = $customer;
+    }
+
+    public function getCustomer()
+    {
+        return $this->_customer;
+    }
+
+    public function getMailChimpTags()
+    {
+        return $this->_mailChimpTags;
+    }
+
+    public function addMailChimpTag($key, $value)
+    {
+        $this->_mailChimpTags[$key] = $value;
+    }
+
+    public function getMailChimpTagValue($key)
+    {
+        return $this->_mailChimpTags[$key];
+    }
+
+    /**
+     * @return array|null
+     * @throws Mage_Core_Exception
+     */
+    public function buildMailChimpTags()
+    {
+        $helper = $this->getMailchimpHelper();
+        $storeId = $this->getStoreId();
+        $mapFields = $helper->getMapFields($storeId);
+        $maps = $this->unserializeMapFields($mapFields);
+
+        $attrSetId = $this->getEntityAttributeCollection()
+            ->setEntityTypeFilter(1)
+            ->addSetInfo()
+            ->getData();
+
+        $this->saveLastOrderInSession();
+
+        foreach ($maps as $map) {
+            $customAtt = $map['magento'];
+            $chimpTag = $map['mailchimp'];
+            if ($chimpTag && $customAtt) {
+                $key = strtoupper($chimpTag);
+
+                if (is_numeric($customAtt)) {
+                    $this->buildCustomerAttributes($attrSetId, $customAtt, $key);
+                } else {
+                    $this->buildCustomizedAttributes($customAtt, $key);
+                }
+            }
+        }
+
+        $newVars = $this->getNewVarienObject();
+        $this->dispatchEventMergeVarAfter($newVars);
+
+        if ($newVars->hasData()) {
+            $this->mergeMailchimpTags($newVars->getData());
+        }
+    }
+
+    protected function getNewVarienObject()
+    {
+        return new Varien_Object;
+    }
+    /**
+     * @param $subscriberEmail
+     * @return mixed
+     * @throws Mage_Core_Exception
+     */
+    protected function saveLastOrderInSession()
```

Comment:
Remove unnecessary function and call directly getLastOrderByEmail when it is required.